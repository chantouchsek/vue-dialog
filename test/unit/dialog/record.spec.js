
import Dialog from '../../../src/dialog'

test('Check is not isNewRecord', async () => {
  const dlg = new Dialog({
    template: '<p>{{ isNewRecord }}</p>',
    primaryKey: 'id'
  })
  await dlg.show({ id: 123 })
  expect(dlg.element.innerHTML).toMatchSnapshot()
  dlg.close()
})

test('Check is isNewRecord', async () => {
  const dlg = new Dialog({
    template: '<p>{{ isNewRecord }}</p>',
    primaryKey: 'id'
  })
  await dlg.show()
  expect(dlg.element.innerHTML).toMatchSnapshot()
  dlg.close()
})

test('Check is isNewRecord undefined', async () => {
  const dlg = new Dialog({
    template: '<p></p>'
  })
  await dlg.show()
  expect(dlg.vm.isNewRecord).toBeUndefined()
  dlg.close()
})

test('Check parameters', async () => {
  const dlg = new Dialog({
    template: '<p></p>',
    primaryKey: 'id'
  })
  const params = { id: 1, name: 'foo' }
  await dlg.show(params)
  expect(dlg.vmd.$parameters).toMatchObject(params)
  dlg.close()
})
