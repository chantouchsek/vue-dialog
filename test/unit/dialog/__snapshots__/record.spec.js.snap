// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check is isNewRecord 1`] = `"<p>true</p>"`;

exports[`Check is not isNewRecord 1`] = `"<p>false</p>"`;
